### PR TITLE
Add error message component

### PR DESCRIPTION
**Pull request by stephaniehobson**
## Description

Add error message component and update documentation associated with form validation.

- [x] I have documented this change in the design system.
- [x] I have recorded this change in `CHANGELOG.md`.

### Issue

Error messages are defined in https://github.com/mozilla/protocol/issues/430#issuecomment-556399694

### Testing

- http://localhost:3000/demos/forms/validation.html
- http://localhost:3000/patterns/molecules/forms.html#message
- http://localhost:3000/patterns/atoms/forms.html#error-list

### CODE REVIEW

**Review by craigcook — COMMENTED** [review]
This all looks great, my only nitpick is that the red background might be too much. Maybe it's just my aging eyes but I find it hard to read the small white text on the bright red:

image

We could go a little darker for better contrast maybe?

**Comment by stephaniehobson** [comment]
Updated.

File: src/assets/sass/protocol/includes/forms/_lib.scss [hunk 1]
```diff
@@ -80,3 +81,29 @@ $line-height-shim: 0.15em; // two elements with text appear to have more space b
...
+    margin: 2px 0 0;
+    max-width: 100%;
+    padding: $spacing-xs #{$spacing-sm + 2px};
+    position: relative;
+}
+
+@mixin form-msg-pointer() {
+    @include bidi(((left, 6px, right, auto),));
+    border: 7px solid transparent;
+    border-bottom: 8px solid $form-red;
```
Comment (craigcook) [inline src/assets/sass/protocol/includes/forms/_lib.scss:102]: The pointer should be the same color as the background.

**Review by craigcook — APPROVED** [review]
The darker red is definitely better contrast, just need to update the little pointer thingie to match. r+wc! 🥠

**Comment by stephaniehobson** [comment]
Updated.

**Pull request merged**